src/wallet/rpcwallet.cpp

```diff
@@ -3990,10 +3990,17 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
+    if (!fEnableZGetBalance)
+        throw runtime_error(
+            "z_getbalance is DEPRECATED and will be removed in a future release\n\n"
+            "Use z_getbalanceforaccount instead, or restart with `-allowdeprecated=z_getbalance`\n"
```

Comment:
`z_getbalanceforaccount` can't replace this API, as it can't show balances for legacy shielded addresses. And we should direct transparent address users to `getbalance`.
```suggestion
            "Use z_getbalanceforviewingkey or getbalance instead, or restart with `-allowdeprecated=z_getbalance`\n"
```